packages/ui/src/ActionDialog/ActionDialog.tsx

```diff
@@ -0,0 +1,59 @@
+import React from "react";
+
+import Button from "../Button";
+import { ButtonSize } from "../Button/Button";
+
+export interface ActionDialogProps {
+  title: string;
+  setOpen: (open: boolean) => void;
+  onConfirm: () => void;
+  confirmLabel?: string;
+  cancelLabel?: string;
+}
+
+const ActionDialog: React.FC<ActionDialogProps> = ({
+  title,
+  setOpen,
+  onConfirm = () => {},
+  confirmLabel = "Confirm",
+  cancelLabel = "Cancel",
+  children,
+}) => {
+  const handleConfirm = () => {
+    setOpen(false);
+    onConfirm();
+  };
+
+  const handleCancel = () => {
+    setOpen(false);
+  };
+
+  return (
+    <div className="flex flex-col max-w-sm items-start gap-y-4 p-5 rounded-md bg-white">
+      <div>
+        <h3 className="text-base leading-6 font-semibold text-red-700">
+          {title}
+        </h3>
+      </div>
+      <div>{children}</div>
+      <div className="flex space-x-2">
```

Comment:
I would add `w-full` and `justify-right` as that seems more natural for me visually, but do your judgement here.